feature/bookmarks/src/test/kotlin/com/google/samples/apps/nowinandroid/feature/bookmarks/BookmarksViewModelTest.kt

```diff
@@ -88,6 +90,54 @@ class BookmarksViewModelTest {
         val item = viewModel.feedUiState.value
         assertIs<Success>(item)
         assertEquals(item.feed.size, 0)
+        assertTrue(viewModel.shouldDisplayUndoBookmark)
+
+        collectJob.cancel()
+    }
+
+    @Test
+    fun whenResourceViewed_setResourcesViewed() = runTest {
+        val collectJob = launch(UnconfinedTestDispatcher()) { viewModel.feedUiState.collect() }
+
+        // Give
+        newsRepository.sendNewsResources(newsResourcesTestData)
+        userDataRepository.setNewsResourceBookmarked(newsResourcesTestData[0].id, true)
+        val itemBeforeViewed = viewModel.feedUiState.value
+        check(itemBeforeViewed is Success)
+        check(!itemBeforeViewed.feed.first().hasBeenViewed)
+
+        // When
+        viewModel.setNewsResourceViewed(newsResourcesTestData[0].id, true)
+
+        // Then
+        val item = viewModel.feedUiState.value
+        assertIs<Success>(item)
+        assertTrue(item.feed.first().hasBeenViewed)
+
+        collectJob.cancel()
+    }
+
+    @Test
+    fun whenUndoBookmarkRemoval_thenBookmarkIsRestored() = runTest {
+        val collectJob = launch(UnconfinedTestDispatcher()) { viewModel.feedUiState.collect() }
+
+        // Give
+        newsRepository.sendNewsResources(newsResourcesTestData)
+        userDataRepository.setNewsResourceBookmarked(newsResourcesTestData[0].id, true)
+        viewModel.removeFromSavedResources(newsResourcesTestData[0].id)
+        check(viewModel.shouldDisplayUndoBookmark)
+        val itemBeforeUndo = viewModel.feedUiState.value
+        check(itemBeforeUndo is Success)
+        check(itemBeforeUndo.feed.isEmpty())
+
+        // When
+        viewModel.undoBookmarkRemoval()
+
+        // Then
+        assertFalse(viewModel.shouldDisplayUndoBookmark)
+        val item = viewModel.feedUiState.value
+        assertIs<Success>(item)
+        assertEquals(item.feed.size, 1)
```

Comment:
in `assertEquals`, the first argument is the `expected`, and the second is the `actual`:

```suggestion
        assertEquals(1, item.feed.size)
```

There are 2 other wrong usages in this file.